php/example_code/auto-scaling/AutoScalingService.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#snippet-start:[php.example_code.auto-scaling.service]
+namespace AutoScaling;
+
+use Aws\AutoScaling\AutoScalingClient;
+use Aws\AutoScaling\Exception\AutoScalingException;
+use AwsUtilities\AWSServiceClass;
+use Exception;
+
+class AutoScalingService extends AWSServiceClass
+{
+    protected AutoScalingClient $autoScalingClient;
+
+    public function __construct($autoScalingClient)
+    {
+        $this->autoScalingClient = $autoScalingClient;
+    }
+
+    #snippet-start:[php.example_code.auto-scaling.service.createAutoScalingGroup]
+    public function createAutoScalingGroup(
+        $autoScalingGroupName,
+        $availabilityZones,
+        $minSize,
+        $maxSize,
+        $launchTemplateId
+    ) {
+        return $this->autoScalingClient->createAutoScalingGroup([
+            'AutoScalingGroupName' => $autoScalingGroupName,
+            'AvailabilityZones' => $availabilityZones,
+            'MinSize' => $minSize,
+            'MaxSize' => $maxSize,
+            'LaunchTemplate' => [
+                'LaunchTemplateId' => $launchTemplateId,
+            ],
+        ]);
+    }
+    #snippet-end:[php.example_code.auto-scaling.service.createAutoScalingGroup]
+
+    #snippet-start:[php.example_code.auto-scaling.service.updateAutoScalingGroup]
+    public function updateAutoScalingGroup($autoScalingGroupName, $args)
+    {
+        if (array_key_exists('MaxSize', $args)) {
+            $maxSize = ['MaxSize' => $args['MaxSize']];
+        } else {
+            $maxSize = [];
+        }
```

Comment:
One of the few places I find ternaries might be cleaner?

```
$maxSize = array_key_exists('MaxSize', $args)
    ? ['MaxSize' => $args['MaxSize']
    : [];
    ```

Reply:
I would normally actually go
```
$maxSize = [];
if(array_key_exists('MaxSize', $args)) {
    $maxSize = ...;
}

```
But I was going for clarity here, and if/else is the easiest to read.